### PR TITLE
dedicated server management rewrite

### PR DESCRIPTION
**Pull request by turtletowerz**
### What is the new feature?
Overhaul of the current dedicated server management script and associated resources.

### Why should this be part of tModLoader?
Simplifies the user experience by dividing setup steps into commands and gives command flags more appropriate names. It also fixes some bugs in the original management script such as incorrect Github version updating and legacy versions not being properly supported.

### Are there alternative designs?
The original design was sufficient but it had some issues which are noted above.

### Sample usage for the new feature
Included in the README

Additionally, the Docker container is currently not functional due to (presumably) a TML bug. This is noted in the README and will be fixed when a fix is available.

### CODE REVIEW

File: patches/tModLoader/Terraria/release_extras/DedicatedServerUtils/manage-tModLoaderServer.sh [hunk 1]
```diff
@@ -38,230 +36,251 @@ function update_script {
 		echo "No new script updates"
 	fi
 
-	exit 0
+	exit
+}
+
+function verify_steamcmd {
```
Comment (Solxanich) [inline patches/tModLoader/Terraria/release_extras/DedicatedServerUtils/manage-tModLoaderServer.sh]: Please add more comments to this function.

Consider rename to 'VerifyDownloadToolDependencies' or something to acknowledge ZIP

File: patches/tModLoader/Terraria/release_extras/DedicatedServerUtils/manage-tModLoaderServer.sh [hunk 2]
```diff
@@ -38,230 +36,251 @@ function update_script {
...
+		local latest_release
+		latest_release=$({
+			curl -s "$release_url" 2>/dev/null || wget -q -O- "$release_url";
+		} | grep '"tag_name":' | sort | tail -1 | sed -E 's/.*"([^"]+)".*/\1/') # Get latest release from github's api
+		echo "$latest_release" # So functions calling this can consume the result since you can't return strings in bash :)
+	fi
 }
 
 # Takes version number as first parameter
 function down_release {
```
Comment (Solxanich) [inline patches/tModLoader/Terraria/release_extras/DedicatedServerUtils/manage-tModLoaderServer.sh]: Please expand to download_release

File: patches/tModLoader/Terraria/release_extras/DedicatedServerUtils/manage-tModLoaderServer.sh [hunk 3]
```diff
@@ -350,79 +364,98 @@ while [[ $# -gt 0 ]]; do
...
+		chmod u+x start-tModLoaderServer.sh
+		./start-tModLoaderServer.sh -nosteam
+		;;
+	*)
+		echo "Invalid Command: $1"
+		print_help
+		;;
+esac
+
+# TODO: Disable this in docker?
```
Comment (Solxanich) [inline patches/tModLoader/Terraria/release_extras/DedicatedServerUtils/manage-tModLoaderServer.sh]: Lets disable this in docker for now. 
Best to play safe

**Review by Solxanich — COMMENTED** [review]
Some brief cleanups.

Great work updating this; 
I'll have to announce this change for players / server owners in itself since it is a few changes

**Comment by turtletowerz** [comment]
applied all of your changes, thanks for the feedback!

**Pull request merged**